java/vector/src/main/java/org/apache/arrow/vector/table/BaseTable.java

```diff
@@ -266,6 +270,48 @@ FieldVector getVector(int columnIndex) {
     return fieldVectors.get(columnIndex);
   }
 
+
+  /**
+   * Returns a copy of the vector with the given name, or throws IllegalArgumentException if the name is not found.
+   * Names are case-sensitive.
+   *
+   * @param columnName The name of the vector to copy
+   * @return A copy of the Vector with the given name
+   * @throws IllegalArgumentException if the name is not the name of a vector in the table.
```

Comment:
nit: the exception class is different from what's actually thrown

Reply:
I will fix the exception class issue. Thanks